### ISSUE TITLE
Could not access file: EACCES: permission denied, mkdir '/data/db/.mongodb'

### ISSUE THREAD

**Issue by zivfuture** [mongodb/mongodb-kubernetes-operator#1141]
**What did you do to encounter the bug?**
manager.yaml:
![image](https://user-images.githubusercontent.com/49356505/197373641-eb5815b0-0c76-4272-9845-e539236eaf5a.png)

```yaml
#mongodb.yaml:

apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: mongodb
  namespace: db
spec:
  version: "6.0.2"
  type: ReplicaSet
  members: 1
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: admin
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: mongodb
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: mongo-admin
  additionalMongodConfig:
    systemLog:
      destination: file
      path: /var/log/mongodb-mms-automation/mongodb.log
      logAppend: true
    storage:
      journal:
          enabled: true
    net:
      bindIp: 0.0.0.0
      port: 27017
  statefulSet:
    spec:
      serviceName: mongodb
      selector: {}
      template:
        metadata:
          labels:
            app: mongodb
        spec:
          securityContext:
            runAsUser: 2000
            runAsGroup: 2000
            fsGroup: 2000
            runAsNonRoot: true
          containers:
          - name: mongod
            imagePullPolicy: IfNotPresent  #default is always
          - name: mongodb-agent
            imagePullPolicy: IfNotPresent  #default is always
      volumeClaimTemplates:
        - metadata:
            name: data-volume
            namespace: db
          spec:
            accessModes: [ "ReadWriteOnce", "ReadWriteMany" ]
            resources:
              requests:
                storage: 5Gi
        - metadata:
            name: logs-volume
            namespace: db
          spec:
            accessModes: [ "ReadWriteOnce", "ReadWriteMany" ]
            resources:
              requests:
                storage: 2Gi

```

**What did you expect?**
works fine

**What happened instead?**
```shell
I have no name!@mongodb-0:/$  mongosh "mongodb://admin:xxx@ip:port/admin?authSource=admin"
Warning: Could not access file: EACCES: permission denied, mkdir '/data/db/.mongodb'
```

**Screenshots**
run "kubect exec it" into container and ls files:
![image](https://user-images.githubusercontent.com/49356505/197373754-dd16cf86-89b1-49e8-91ac-9fbef5dd2e93.png)
you can see the directory  has wrong uid and groups and wrong permission: `rwxr-xr-x`.

cat /etc/passwd:
![image](https://user-images.githubusercontent.com/49356505/197373800-2259e6bc-f152-46c6-aeb8-6a9de368eaf7.png)

**Operator Information**
 - Operator Version
 - MongoDB Image used

**Kubernetes Cluster Information**
 - Version :  v1.24.1

Where am i missed?

**Comment by mircea-cosbuc** [comment]
I will close this ticket for now, please re-open if the provided information is not sufficient for the use-case.

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
